Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeContext.mm

```diff
@@ -58,21 +58,25 @@
 {
     for (auto& layer : m_livePlatformLayers.values())
         layer->clearContext();
+    m_livePlatformLayers.clear();
```

Comment:
Why?

Reply:
These containers contain CheckedPtr to PlatformCALayerRemote objects, which cannot outlive the PlatformCALayerRemote objects themselves (for security reasons). However, one of the data members m_layerPool may destroy those layers upon destruction, meaning that the CheckedPtrs would point to a dead object during the destruction process (though not for long). This would trip our RELEASE_ASSERT() in CanMakeCheckedPtr.

For this reason, I have to make sure these containers are actually empty before we start destroying data members. Destruction ordering is important here. This was caught by our layout tests.

Reply:
I added a code comment about this.